research/format.markdown

```diff
@@ -0,0 +1,138 @@
+
+**\-**     **time** : This SHOULD be a time in the format of hh:mm:ss.
+
+**\-**     **utc-millisec** : This SHOULD be the difference, measured in milliseconds, between the specified time and midnight, 00:00 of January 1, 1970 UTC. The value SHOULD be a number (integer or float).
+
+**\-**     **regex** : A regular expression, following the regular expression specification from ECMA 262/Perl 5.
+
+**\-**     **color** : This is a CSS color (like "#FF0000" or "red"), based on CSS 2.1 (Hickson, I., Lie, H., Çelik, T., and B. Bos, “Cascading Style Sheets Level 2 Revision 1 (CSS 2.1) Specification,” July 2007.) [W3C.CR‑CSS21‑20070719].
+
+**\-**     **style** : This is a CSS style definition (like "color: red; backgroundcolor:#FFF"), based on CSS 2.1 (Hickson, I., Lie, H., Çelik, T., and B. Bos, “Cascading Style Sheets Level 2 Revision 1 (CSS 2.1) Specification,” July 2007.) [W3C.CR‑CSS21‑20070719].
+
+**\-**     **phone** : This SHOULD be a phone number (format MAY follow E.123).
+
+### Changes: 
+
+| Attribute   | Previous Description                                                                      | PresentDescription           |
+|-------------|-------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------|
+| date-time   | This SHOULD be a date in ISO 8601 format of YYYY-MMDDThh:mm:ssZ in UTC time. This is... | A string instance is valid against this attribute if it is a valid date representation as defined by RFC 3339, section 5.6[RFC3339].                                  |
+| URI         | Should be a uri                                                                           | A string instance is valid against this attribute if it is a valid URI, according to [RFC3986].           |
+| email       | Should be email                                                                           | A string instance is valid against this attribute if it is a valid Internet email address as defined by RFC 5322, section 3.4.1 [RFC5322].                         |
+| ip-address  | Should be ip v 4 address                                                                 | ipv4: A string instance is valid against this attribute if it is a valid representation of an IPv4 address according to the "dotted-quad" ABNF syntax as defined in RFC 2673, section 3.2 [RFC2673]. |
+| ipv6        | Should be ip v 6 address                                                                 | A string instance is valid against this attribute if it is a valid representation of an IPv6 address as defined in RFC 2373, section 2.2 [RFC2373].                 |
+| host-name   | This SHOULD be a host-name                                                               | A string instance is valid against this attribute if it is a valid representation for an Internet host name, as defined by RFC 1034, section 3.1 [RFC1034].           |
+
```

Comment:
Why are there consistent blank lines? Between two pieces of text, there should always be one and only one blank line

Reply:
I thought it was good to have more lines for different dialects. Like Python..!?

Reply:
Not really needed, and the blank lines will get ignored by the markdown renderer anyway